File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams = defaultdict(list)
        for char in strs:
            key = "".join(sorted(char))

            anagrams[key].append(char)
        return list(anagrams.values())

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> PairMap;
        vector<vector<string>> result;

        for(string s:strs)
        {
            string Sword = s;
            sort(Sword.begin(),Sword.end());
            PairMap[Sword].push_back(s);
        }

        for(auto& i:PairMap)
        {
            result.push_back(i.second);
        }

        return result;
    }
};